### PR TITLE
FARO-1339-Velocity-Log-Changes

### PR DESCRIPTION
**Pull request by raaayden**
made changes to the velocity log to incorporate member data

### CODE REVIEW

File: Clients/FARO/Analytics Script/velocity/faro_velocity_log.sql [hunk 1]
```diff
@@ -1,5 +1,5 @@
 --------------------------------------------------------------
--------------------------- VELOCITY --------------------------
+-------------------------- Activity --------------------------
 --------------------------------------------------------------
 
 CREATE OR REPLACE TABLE `x-marketing.faro.db_activity_log` AS
```
Comment (izztsfn) [inline Clients/FARO/Analytics Script/velocity/faro_velocity_log.sql:5]: Replace with Insert statement instead

File: Clients/FARO/Analytics Script/velocity/faro_velocity_log.sql [hunk 2]
```diff
@@ -49,99 +49,28 @@ WITH activity AS (
...
-    AND contacts.field IN ('Waterfall_Stage__c', 'Status')
-    AND (contacts.field != 'Status' OR contacts.newvalue = 'No Current Interest-Recycled')
-  -- QUALIFY ROW_NUMBER() OVER(PARTITION BY contacts.id ORDER BY contacts.createddate) = 1
-)
-SELECT * FROM prospect_data;
+--------------------------------------------------------------
+------------------------ Member Stat -------------------------
+--------------------------------------------------------------
 
+CREATE OR REPLACE TABLE `x-marketing.faro.db_member_log` AS
```
Comment (izztsfn) [inline Clients/FARO/Analytics Script/velocity/faro_velocity_log.sql:56]: Replace with Insert statement instead

File: Clients/FARO/Analytics Script/velocity/faro_velocity_log.sql [hunk 3]
```diff
@@ -49,99 +49,28 @@ WITH activity AS (
...
+  member.status AS _member_status,
+  member.lastmodifieddate AS _last_modified_date
+FROM `x-marketing.faro_salesforce.CampaignMember` member
+LEFT JOIN `x-marketing.faro_salesforce.Campaign` campaign
+  ON campaign.id = member.campaignid;
 
+--------------------------------------------------------------
+-------------------------- Velocity --------------------------
+--------------------------------------------------------------
 CREATE OR REPLACE TABLE `x-marketing.faro.db_velocity_log` AS
```
Comment (izztsfn) [inline Clients/FARO/Analytics Script/velocity/faro_velocity_log.sql:71]: Replace with Insert statement instead

File: Clients/FARO/Analytics Script/velocity/faro_velocity_log.sql [hunk 4]
```diff
@@ -241,4 +166,33 @@ FROM stages_data
 LEFT JOIN activity
   ON activity._prospect_id = stages_data._prospect_id
   AND activity._old_new = CONCAT(stages_data._old_value,stages_data._new_value)
-  AND activity._created_date = stages_data._created_date;
\ No newline at end of file
+  AND activity._created_date = stages_data._created_date;
+
+------------------------- RECENT LOG -------------------------
+CREATE OR REPLACE TABLE `x-marketing.faro.db_velocity_recent` AS
```
Comment (izztsfn) [inline Clients/FARO/Analytics Script/velocity/faro_velocity_log.sql:172]: Replace with Insert statement instead

File: Clients/FARO/Analytics Script/velocity/faro_velocity_log.sql [hunk 5]
```diff
@@ -241,4 +166,33 @@ FROM stages_data
...
+  velocity.*,
+  member_log.*
+FROM `x-marketing.faro.db_velocity_log` velocity
+LEFT JOIN `x-marketing.faro.db_member_log` member_log
+  ON velocity._prospect_id = member_log._leadorcontactid
+  AND member_log._last_modified_date 
+  BETWEEN velocity._created_date AND velocity._next_change_date;
+
+------------------------- MEMBER STAT -------------------------
+CREATE OR REPLACE TABLE `x-marketing.faro.db_member_stat` AS
```
Comment (izztsfn) [inline Clients/FARO/Analytics Script/velocity/faro_velocity_log.sql:183]: Replace with Insert statement instead

**Review by izztsfn — CHANGES_REQUESTED** [review]
@raaayden the formatting looks good. Just need to update the Create statements to use Replace instead for best practices. Thankssss

**Comment by raaayden** [comment]
hello @izztsfn  i have updated the script can you please check it again 👍

**Review by izztsfn — APPROVED** [review]
Looks good!

**Pull request merged**